spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala

```diff
@@ -303,8 +308,14 @@ object TypeConverter {
   }
 
   implicit object SqlDateConverter extends NullableTypeConverter[java.sql.Date] {
+
+    def subtractTimeZoneOffset(millis: Long) = millis - defaultTimezone.getOffset(millis)
+
     def targetTypeTag = SqlDateTypeTag
-    def convertPF = DateConverter.convertPF.andThen(d => new java.sql.Date(d.getTime))
+
+    def convertPF = PartialFunction[Any, java.sql.Date]{ case x: LocalDate =>
```

Comment:
My understanding is that the `case x: LocalDate => ...` will be inner and `DateConverter.convertPF` will be outer. So if the inner function fails, the it will fail, because it has no alternative. Also, look that the type of the later function - the argument type is the same as return type of the first function.

Consider this:
```
scala> val pf1 = PartialFunction[Any, String] {
  case x: String => s"String $x"
  case x: Int => s"Int $x"
}
     |      |      | pf1: PartialFunction[Any,String] = <function1>

scala> val pf0 = PartialFunction[Any, String] {
  case x: Int => s"*** $x"
}
     |      | pf0: PartialFunction[Any,String] = <function1>

scala> val pf = pf0 andThen pf1
pf: PartialFunction[Any,String] = <function1>

scala> pf.apply(1)
res3: String = String *** 1

scala> pf.apply("a")
scala.MatchError: a (of class java.lang.String)
```

I think you should use `orElse` as you said.